code/modules/mob/living/simple_animal/abnormality/_tools/zayin/touch.dm

```diff
@@ -26,55 +26,57 @@
 
 	if(round_end)
-		SSpersistence.rounds_since_button_pressed = ROUNDCOUNT_BUTTON_PRESSED
-		for(var/obj/structure/sign/button_counter/sign as anything in GLOB.map_button_counters)
-			sign.update_count(ROUNDCOUNT_BUTTON_PRESSED)
+		bastards += user.ckey
 		RoundEndEffect(user)
 	else
+		breaching_bastards += user.ckey
 		BreachEffect(user)
 
 /obj/structure/toolabnormality/touch/proc/RoundEndEffect(mob/living/carbon/human/user)
-	bastards += user.ckey
-	if(do_after(user, 45 SECONDS))
-		SSticker.SetRoundEndSound('sound/abnormalities/donttouch/end.ogg')
-		SSticker.force_ending = 1
-		var/ending = pick(1,2)
-		switch(ending)
-			if(1)
-				for(var/mob/M in GLOB.player_list)
-					if(isnewplayer(M))
-						continue
-					flash_color(M, flash_color = COLOR_RED, flash_time = 150)
-					M.playsound_local(M, pick('sound/abnormalities/donttouch/kill.ogg', 'sound/abnormalities/donttouch/kill2.ogg'), 50, FALSE)
-					if(ishuman(M))
-						var/mob/living/carbon/human/H = M
-						H.gib()
-			if(2)
-				for(var/mob/M in GLOB.player_list)
-					if(isnewplayer(M))
-						continue
-					flash_color(M, flash_color = COLOR_RED, flash_time = 150)
-					M.playsound_local(M, 'sound/abnormalities/donttouch/panic.ogg', 50, FALSE)
-					if(ishuman(M))
-						var/mob/living/carbon/human/H = M
-						H.apply_damage(5000, WHITE_DAMAGE, null, null, spread_damage = TRUE)//You cannot escape.
-	else
-		user.gib() //lol, idiot.
+	if(!do_after(user, 60 SECONDS))
+		user.gib() // lol, idiot.
+		return
 
-/obj/structure/toolabnormality/touch/proc/BreachEffect(mob/living/carbon/human/user)
-	breaching_bastards += user.ckey
-	if(do_after(user, 45 SECONDS))
-		for(var/obj/structure/toolabnormality/clock/backclock in world.contents) //prevents an exploit
-			backclock.clock_cooldown = backclock.clock_cooldown_time + world.time
+	SSticker.SetRoundEndSound('sound/abnormalities/donttouch/end.ogg')
+	SSticker.force_ending = 1
+	SSpersistence.rounds_since_button_pressed = ROUNDCOUNT_BUTTON_PRESSED
+	for(var/obj/structure/sign/button_counter/sign as anything in GLOB.map_button_counters)
+		sign.update_count(ROUNDCOUNT_BUTTON_PRESSED)
+
+	if(prob(50)) // Randomly choose to eighter gib or insane everyone
 		for(var/mob/M in GLOB.player_list)
 			if(isnewplayer(M))
 				continue
-			flash_color(M, flash_color = COLOR_RED, flash_time = 30)
-			M.playsound_local(M, 'sound/abnormalities/donttouch/breach.ogg', 50, FALSE)
-		for(var/datum/abnormality/A in SSlobotomy_corp.all_abnormality_datums)
-			A.qliphoth_change(-A.qliphoth_meter) // Down to 0
+			flash_color(M, flash_color = COLOR_RED, flash_time = 150)
+			M.playsound_local(M, pick('sound/abnormalities/donttouch/kill.ogg', 'sound/abnormalities/donttouch/kill2.ogg'), 50, FALSE)
+			if(ishuman(M))
+				var/mob/living/carbon/human/H = M
+				H.gib()
+	else
+		for(var/mob/M in GLOB.player_list)
+			if(isnewplayer(M))
```

Comment:
What is the point of this?
Shouldn't it check beforehand if they're a new player?

Reply:
explained in discord, isnewplayer() checks if the person is in the lobby mob.